elixir/apps/domain/test/domain/change_logs_test.exs

```diff
@@ -130,4 +131,231 @@ defmodule Domain.ChangeLogsTest do
       end
     end
   end
+
+  describe "truncate/2" do
+    setup do
+      account1 = Fixtures.Accounts.create_account()
+      account2 = Fixtures.Accounts.create_account()
+      %{account1: account1, account2: account2}
+    end
+
+    test "deletes change logs before cutoff for specific account", %{
+      account1: account1
+    } do
+      now = DateTime.utc_now()
+
+      # Create some old records (before cutoff)
+      old_attrs = [
+        %{
+          account_id: account1.id,
+          lsn: 1,
+          table: "resources",
+          op: :insert,
+          old_data: nil,
+          data: %{"id" => "1"},
+          vsn: 1,
+          inserted_at: now
+        },
+        %{
+          account_id: account1.id,
+          lsn: 2,
+          table: "resources",
+          op: :update,
+          old_data: %{"id" => "1", "name" => "old"},
+          data: %{"id" => "1", "name" => "new"},
+          vsn: 1,
+          inserted_at: now
+        }
+      ]
+
+      assert {2, nil} = bulk_insert(old_attrs)
+
+      # Sleep to ensure time difference
+      Process.sleep(1)
```

Comment:
Rather than `sleep`, would it be possible to just set DateTimes that are further back in time?

Reply:
Fixed